### PR TITLE
Add support for tracking multiple players' game state.

### PR DESCRIPTION
**Pull request by arensonjr**
- Site can now track cities/payoffs for 1-5 players
  - Each player has their own colored div (chosen in settings)
  - Number of players can be changed/chosen in settings as well
  - Modified URL params to be "&players=<json-string>&players=<json-string>" so that we could persist cities for each player
- Refactored $watchGroup() code to $watch() each variable individually instead
  - Impossible to "deep-watch" the list of players unless I watch it on its own
  - Also, tracking numPlayers would have produced some spaghetti code

Feel free to nitpick this as much as you want. I'm mostly new to Javascript and completely new to Angular, so if you're wondering why I didn't do X/Y/Z, it's probably because I didn't know it was an option. Thanks!

### CODE REVIEW

**Comment by sir-buckyball** [comment]
Out of curiosity, how are you using this app in game play?

When we played, we used a couple of android phones (not everyone had one), so some players helped keep track for other players. The way we dealt with multiple players on a single phone was to simply open up multiple tabs (which for me show up as android application windows), and to color the background so it was easy to find which window to switch to.

What I've struggled with is whether the hassle of switching windows is worth the UX/code complexity of multiple players on a single page. What are your thoughts?

**Comment by arensonjr** [comment]
I find switching tabs terribly onerous on a phone -- multitasking is not something I'm fond of on Android.

My experience using V1 of this app was that, like you said, not everyone had a phone to use. The easiest answer for us was to go get an extra phone and bring up another copy of the webpage. My goal in making this was to support the extra/new players without needing extra devices (or tab-switching every 30 seconds on mine).

I would have considered the UX complex if it weren't for your great idea of colored backgrounds -- I kept that, except now instead of coloring the entire page, I'm just coloring a solid rectangle where the player's city data is (sample: http://arensonjr.github.io/boxcars_payoff). The one remaining UX bit that I'm not personally fond of is that the settings menu covers up the bottom 10% of the screen (starts to be a problem with 3 players on my phone).

File: index.html [hunk 1]
```diff
@@ -569,70 +634,95 @@
...
+      $scope.updateNumPlayers();
     };
-    $scope.$on("$locationChangeSuccess", doInit);
     doInit();
   });
+
+  /**
+   * Allow representing models directly as numbers, not strings.
+   */
+  app.directive('convertToNumber', function() {
```
Comment (sir-buckyball) [inline index.html]: Is this directive actually used? I don't see a reference.

**Comment by sir-buckyball** [comment]
Thanks for the feedback.lets go forward on this. I'll look through the code and make comments later today. Looking at your demo, I think the most critical change is to not have the settings overlap the players. It's very awkward now the the page has substantial height

File: index.html [hunk 2]
```diff
@@ -481,45 +485,102 @@
...
-        $scope.payoff = $scope.data.payoff[orig][dest];
+      // Update the location history
+      $location.search("map", map);
+    });
+
+    /**
+     * Watch for player city changes.
+     */
+    $scope.$watch("players", function(players) {
+      $scope.updatePlayers();
```
Comment (sir-buckyball) [inline index.html]: The updatePlayers function takes an array of player objects, which you aren't passing in here.

Comment (sir-buckyball) [inline index.html]: If you put this code after your definition of updatePlayers, it wouldn't be necessary to wrap it in a function. eg., you could have: $scope.$watch("players", $scope.updatePlayers, true);

File: index.html [hunk 3]
```diff
@@ -481,45 +485,102 @@
...
+        }
+      } else {
+        for (var i = 0; i > playerDelta; i--) {
+          $scope.players.pop();
+        }
+      }
+    };
+
+    /**
+     * Human name --> CSS name
```
Comment (sir-buckyball) [inline index.html]: This comment is backwards. The keys in this map are the CSS names, and the values are the piece colors in the game.

http://www.w3schools.com/cssref/css_colornames.asp

File: index.html (hunk 2 above)
Comment (arensonjr) [inline index.html]: The function parameter got refactored away, and I missed deleting it here (and, of course, Javascript doesn't complain). Whoops.

Thanks for the tip -- done.

File: index.html (hunk 3 above)
Comment (arensonjr) [inline index.html]: Updated to be slightly more eloquent.

File: index.html (hunk 1 above)
Comment (arensonjr) [inline index.html]: Nope -- I originally had it set up on the numPlayers select box, but that got refactored away. Deleted.

**Comment by sir-buckyball** [comment]
Thanks for making the changes.

**Comment by arensonjr** [comment]
I'll work on a concept for getting the settings UI out of the way of the
extra players' info...

**Pull request merged**